### PR TITLE
[pipelining] Stress test schedules with multi iters

### PR DESCRIPTION
**Pull request by kwen2501**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #127479
* __->__ #127475

cc @mrshenli @pritamdamania87 @zhaojuanmao @satgera @gqchen @aazzolini @osalpekar @jiayisuse @H-Huang @awgu @penguinwu @fegin @XilunWu @wanchaol @fduwjj @wz337 @tianyu-l @wconstab @yf225 @chauhang @d4l3k

### CODE REVIEW

File: test/distributed/pipelining/test_schedule.py [hunk 1]
```diff
@@ -52,6 +52,46 @@ def setUpClass(cls):
...
+            pipe,
+            self.rank,
+            device=self.device,
+        )
+
+        # Attach to a schedule
+        schedule = ScheduleClass(stage, chunks, loss_fn=loss_fn)
+
+        # Run
+        for _ in range(20):
```
Comment (wconstab) [inline test/distributed/pipelining/test_schedule.py:86]: nit: would it make more sense to just modify the gradient tests to run multiple iterations, and then the multi-iter test is also testing correctness?

Comment (kwen2501) [reply test/distributed/pipelining/test_schedule.py:86]: I gave it more thoughts but still prefer for these two tests to indicate two different failures. 
This stress test is not for checking numerics, but to catch if our schedules would hang after many iterations.

If any consolidation can be made, I will be happy to make it. At this point I just want to land more tests before release to safeguard library changes.

**Review by wconstab — APPROVED** [review]

**Pull request closed without merging**